### PR TITLE
Add events

### PR DESCRIPTION
**Pull request by crookedneighbor**
### Summary

Adds two events, `paymentMethodRequestable` and `noPaymentMethodRequestable`.

`paymentMethodRequestable` fires when:

* Credit card form goes from invalid to valid
* PayPal is authorized
* when switching from a payment method view where a payment method cannot be requested back to the payment options view where a previously saved payment method can be requested

`noPaymentMethodRequestable` fires when

* Credit card form goes from valid to invalid
* when switching away from a view that has a requestable payment method to one that does not

### Checklist

- [x] Added a changelog entry
- [x] Ran unit tests

### CODE REVIEW

**Comment by crookedneighbor** [comment]
Closes #110 and begins to address #87

File: README.md [hunk 1]
```diff
@@ -184,6 +184,46 @@ braintree.dropin.create({
 
 Payment options omitted from this array will not be offered to the customer.
 
+## Events
+
+### `paymentMethodRequestable` and `noPaymentMethodRequestable`
+
+`paymentMethodRequestable` fires when a payment method goes from being not requestable to requestable and provides the type of payment method (CreditCard, PayPalAccount, etc).
```
Comment (crookedneighbor) [inline README.md]: That's great.

File: README.md [hunk 2]
```diff
@@ -184,6 +184,46 @@ braintree.dropin.create({
 
 Payment options omitted from this array will not be offered to the customer.
 
+## Events
+
+### `paymentMethodRequestable` and `noPaymentMethodRequestable`
+
+`paymentMethodRequestable` fires when a payment method can be retrieved using `requestPaymentMethod`. The event includes an object that provides the type of payment method (CreditCard, PayPalAccount, etc) that is ready to be requested. 
+
+`noPaymentMethodRequestable` fires when a payment method can no longer be retrieved with `requestPaymetnMethod`.
```
Comment (intelliot) [inline README.md]: requestPaymetnMethod -> requestPaymentMethod

File: src/dropin-model.js [hunk 3]
```diff
@@ -24,6 +25,10 @@ DropinModel.prototype = Object.create(EventEmitter.prototype, {
   constructor: DropinModel
 });
 
+DropinModel.prototype.isPaymentMethodRequestable = function () {
+  return Boolean(this._paymentMethodIsRequestable);
```
Comment (skunkworks) [inline src/dropin-model.js]: For my own edification, is this preferrable to `return this._paymentMethodIsRequestable`?

File: test/unit/dropin-model.js [hunk 4]
```diff
@@ -358,4 +358,133 @@ describe('DropinModel', function () {
...
+      this.model._paymentMethodRequestableType = 'TYPE';
+      this.model.setPaymentMethodRequestable({
+        isRequestable: true,
+        type: 'TYPE'
+      });
+
+      expect(this.model._emit).to.not.be.called;
+    });
+
+    it('does not emit when isRequestable state is false and type has changed', function () {
```
Comment (skunkworks) [inline test/unit/dropin-model.js]: This test looks like a copy/paste of the preceding test.

File: src/views/base-view.js [hunk 5]
```diff
@@ -19,6 +19,10 @@ BaseView.prototype.requestPaymentMethod = function (callback) {
   callback(new Error(errors.NO_PAYMENT_METHOD_ERROR));
 };
 
+BaseView.prototype.getPaymentMethod = function () {
+  return this.activeMethodView && this.activeMethodView.paymentMethod;
```
Comment (skunkworks) [inline src/views/base-view.js]: Nice - TIL.

File: src/dropin-model.js (hunk 3 above)
Comment (crookedneighbor) [inline src/dropin-model.js]: We could do `this._paymentMethodIsRequestable` as long as we initialize `this._paymentMethidIsRequestable` as false. I just like the explicit nature of always returning a `Boolean` in this method.

File: test/unit/dropin.js [hunk 6]
```diff
@@ -497,4 +499,44 @@ describe('Dropin', function () {
       }.bind(this));
     });
   });
+
+  describe('isPaymentMethodRequestable', function () {
+    it('returns the value on model.isPaymentMethodRequestable', function () {
```
Comment (skunkworks) [inline test/unit/dropin.js]: `on` -> `of`

**Pull request merged**